src/app/examples/examples.state.ts

```diff
@@ -0,0 +1,24 @@
+import { createFeatureSelector, ActionReducerMap } from '@ngrx/store';
+
+import { todosReducer } from './todos/todos.reducer';
+import { TodosState } from './todos/todos.model';
+import {
+  stockMarketReducer,
+  StockMarketState
+} from './stock-market/stock-market.reducer';
```

Comment:
This was the case why I had to split stuff into separate files. This file is necessary and has to import all reducers and all `<something>State` interfaces to be able to define whole feature module state interface and feature selector.

On the other hand we need feature selector in the reducers to be able to create specific selectors. That caused circular dependency so that the selectors and interfaces had to be extracted into separate files. I don't really see the way out of this, ideas @timdeschryver ?

Reply:
How the NgRx example app is structured (and how we do it at work) is to create the selectors inside `examples.state`. This because you could use selectors from different "reducers", in this project this won't be the case (you won't combine a todo selector with a stock selector). The downside to this approach is that you can get a bloated file which is hard to maintain, because the app/module itself itsn't split up correctly (seperations of concerns).

I might be wrong but if we do this we might put everything in one file again, but on the other hand it feels better to create different files in this project as it will help starters understand the structure better imho.